Sources/System/Process/ProcessID.swift

```diff
@@ -0,0 +1,30 @@
+
+// FIXME(DO NOT MERGE): We need to find a way around this. We want to declare
+// a typealias to a struct from a header, but don't want downstream to import
+// Darwin or the whole header just for that.
+//
+import Darwin
+extension CInterop {
+  public typealias PID = Int32
+  public typealias ProcTaskInfo = proc_taskinfo // FIXME
+  public typealias RUsageInfo = rusage_info_current // FIXME
+}
+
+public struct ProcessID: RawRepresentable, Hashable, Codable {
```

Comment:
It seems to me that this is more of a `Process` (i.e. a system object identified by a pid) than a `ProcessID` - related APIs such as resource usage are really properties of the process more than the ID.

Reply:
This struct is the handle to the process, it's very much like a file descriptor. It's not "safe" to assume it's still valid after a process is torn down, etc.

I've been thinking that a `Process` type would probably be more like an `actor` with interfaces for interprocess communication. Similarly with a proper `File` type (though that might be a moveonly buffered struct). `ProcessID` here is the lower-level systems model of that where we can communicate using descriptors or other forms of IPC.

Reply:
Do you think `Process` will be part of `SystemPackage` or layered on top? If it is part of `SystemPackage` is there a way to make a stub `Process` type so that this type could be `Process.ID`? I'm not sure if this will work for ABI stability (though I also can't say that it won't), but maybe we can create an `enum Process` that we then upgrade to an `actor` (or whatever).